libs/gltfio/src/AssetLoader.cpp

```diff
@@ -1012,9 +1020,17 @@ MaterialInstance* FAssetLoader::createMaterialInstance(const cgltf_material* inp
             break;
     }
 
+    // Check if this material has an extras string.
+    CString extras;
+    const cgltf_size extras_size = inputMat->extras.end_offset - inputMat->extras.start_offset;
+    if (extras_size > 0) {
+        extras = CString(srcAsset->json + inputMat->extras.start_offset, extras_size);
+    }
+
     // This not only creates a material instance, it modifies the material key according to our
     // rendering constraints. For example, Filament only supports 2 sets of texture coordinates.
-    MaterialInstance* mi = mMaterials->createMaterialInstance(&matkey, uvmap, inputMat->name);
+    MaterialInstance* mi = mMaterials->createMaterialInstance(&matkey, uvmap, inputMat->name,
```

Comment:
You would have to document in createMaterialInstance() that the lifetime of `extras` is only valid for the duration of that function call since you allocate the `CString` on the stack.

Reply:
Reasonable point.